File: design-warm-editorial/scripts/convert_images.py
# Convert large PNG images to WebP using Python Pillow
# Run: pip install Pillow && python convert_images.py

import os
import sys
from pathlib import Path

try:
    from PIL import Image
except ImportError:
    print("ERROR: Pillow not installed. Run: pip install Pillow")
    sys.exit(1)

IMAGE_DIR = Path(__file__).parent.parent / "public" / "images" / "real"
SIZE_THRESHOLD = 500 * 1024  # 500KB

def convert():
    png_files = sorted(IMAGE_DIR.glob("*.png"))
    print(f"Found {len(png_files)} PNG files. Converting those over {SIZE_THRESHOLD // 1024}KB...")

    converted = 0
    total_saved = 0

    for f in png_files:
        size = f.stat().st_size
        if size < SIZE_THRESHOLD:
            print(f"  SKIP {f.name} ({size // 1024}KB — under threshold)")
            continue

        webp_path = f.with_suffix(".webp")
        try:
            img = Image.open(f)
            img.save(webp_path, "WEBP", quality=82, method=6)
            new_size = webp_path.stat().st_size
            saved = size - new_size
            pct = round((1 - new_size / size) * 100)
            print(f"  ✓ {f.name} ({size // 1024}KB) → {webp_path.name} ({new_size // 1024}KB) — saved {saved // 1024}KB ({pct}%)")
            f.unlink()
            print(f"    ✗ Deleted original {f.name}")
            converted += 1
            total_saved += saved
        except Exception as e:
            print(f"  ✗ Failed: {f.name} — {e}")

    print(f"\nDone! Converted {converted} files, saved {total_saved // 1024}KB total.")

if __name__ == "__main__":
    convert()

File: design-warm-editorial/scripts/convert-images.mjs
// scripts/convert-images.mjs
// Converts large PNGs to WebP format for better performance
import sharp from 'sharp';
import { readdir, stat, unlink } from 'fs/promises';
import { join, extname, basename } from 'path';

const IMAGE_DIR = join(process.cwd(), 'public', 'images', 'real');
const SIZE_THRESHOLD = 500 * 1024; // 500KB — only convert files larger than this

async function convertImages() {
  const files = await readdir(IMAGE_DIR);
  const pngFiles = files.filter(f => extname(f).toLowerCase() === '.png');

  console.log(`Found ${pngFiles.length} PNG files. Checking sizes...`);

  let converted = 0;
  let totalSaved = 0;

  for (const file of pngFiles) {
    const filePath = join(IMAGE_DIR, file);
    const fileStats = await stat(filePath);

    if (fileStats.size < SIZE_THRESHOLD) {
      console.log(`  SKIP ${file} (${Math.round(fileStats.size / 1024)}KB — under threshold)`);
      continue;
    }

    const webpName = basename(file, '.png') + '.webp';
    const webpPath = join(IMAGE_DIR, webpName);

    try {
      await sharp(filePath)
        .webp({ quality: 82, effort: 6 })
        .toFile(webpPath);

      const webpStats = await stat(webpPath);
      const savedKB = Math.round((fileStats.size - webpStats.size) / 1024);
      const pct = Math.round((1 - webpStats.size / fileStats.size) * 100);

      console.log(`  ✓ ${file} (${Math.round(fileStats.size / 1024)}KB) → ${webpName} (${Math.round(webpStats.size / 1024)}KB) — saved ${savedKB}KB (${pct}%)`);

      // Delete the original PNG after successful conversion
      await unlink(filePath);
      console.log(`    ✗ Deleted original ${file}`);

      converted++;
      totalSaved += fileStats.size - webpStats.size;
    } catch (err) {
      console.error(`  ✗ Failed to convert ${file}:`, err.message);
    }
  }

  console.log(`\nDone! Converted ${converted} files, saved ${Math.round(totalSaved / 1024)}KB total.`);
}

convertImages().catch(console.error);
